packages/server/src/sdk/app/screens/screens.ts

```diff
@@ -1,13 +1,53 @@
+import { context, utils } from "@budibase/backend-core"
+import { Database, DocumentType, Screen, SEPARATOR } from "@budibase/types"
 import { getScreenParams } from "../../../db/utils"
-import { context } from "@budibase/backend-core"
-import { Screen } from "@budibase/types"
+import sdk from "../.."
 
-export async function fetch(db = context.getAppDB()): Promise<Screen[]> {
-  return (
+export async function fetch(
+  db: Database = context.getAppDB()
+): Promise<Screen[]> {
+  const screens = (
     await db.allDocs<Screen>(
       getScreenParams(null, {
         include_docs: true,
       })
     )
   ).rows.map(el => el.doc!)
+
+  if (screens.every(s => s.projectAppId)) {
+    return screens
+  }
+
+  await migrateToProjectApp(screens)
+  return fetch(db)
+}
+
+async function migrateToProjectApp(screens: Screen[]) {
+  const application = await sdk.applications.metadata.get()
+  if (screens.every(s => s.projectAppId)) {
+    return screens
+  }
+
+  const allProjectApps = await sdk.projectApps.fetch()
+  let projectAppId = allProjectApps.find(p => p.name === application.name)?._id
+  if (!projectAppId) {
+    // Forcing the _id to avoid concurrency issues
```

Comment:
What concurrency issues?

Reply:
We could have parallel gets. This is solved using the built-in migrations